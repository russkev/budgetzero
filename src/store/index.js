import Vue from 'vue'
import Vuex from 'vuex'
import account from './modules/account-module'
import budget from './modules/budget-module'
import category from './modules/category-module'
import id from './modules/id-module'
import payee from './modules/payee-module'
import pouchdb from './pouchdb/pouchdb-main'
import reports from './modules/reports-module'
import transaction from './modules/transaction-module'
import user from './modules/user-module'
import pouchdbDelete from './pouchdb/pouchdb-delete'
import pouchdbExport from './pouchdb/pouchdb-export'
import pouchdbFetch from './pouchdb/pouchdb-fetch'
import pouchdbInit from './pouchdb/pouchdb-init'
import pouchdbRemote from './pouchdb/pouchdb-remote'
import { generateId, generateShortId, validateId, compareAscii } from './modules/id-module'
import { updateSingleCategory, defaultCategoryBalance, getCategoryBalance } from './modules/category-module'
import { updateAccountBalances, defaultAccountBalance } from './modules/account-module'
import { calculateTransactionBalanceUpdate } from './modules/transaction-module'
import { validateDate, validateMonth } from '../helper'
import moment from 'moment'

Vue.mixin({
  methods: {
    generateId,
    generateShortId,
    validateId,
    compareAscii,
    defaultAccountBalance,
    defaultCategoryBalance,
    updateAccountBalances,
    updateSingleCategory,
    getCategoryBalance,
    validateDate,
    validateMonth,
    calculateTransactionBalanceUpdate
  }
})

Vue.use(Vuex)

export default new Vuex.Store({
  strict: process.env.NODE_ENV !== 'production',
  modules: {
    account,
    budget,
    category,
    id,
    payee,
    pouchdb,
    reports,
    transaction,
    user,
    pouchdbDelete,
    pouchdbExport,
    pouchdbFetch,
    pouchdbInit,
    pouchdbRemote,
  },
  state: {
    snackbarMessage: '',
    snackbarColor: '',
    snackbar: false,
    sync_state: '',
    selectedBudgetID: null,
    selectedMonth: moment(new Date()).format('YYYY-MM')
  },
  getters: {
    snackbarMessage: (state) => state.snackbarMessage,
    snackbarColor: (state) => state.snackbarColor,
    sync_state: (state) => state.sync_state,
    snackbar: (state) => state.snackbar,
    selectedBudgetID: (state) => state.selectedBudgetID,
    selectedMonth: (state) => state.selectedMonth
  },
  mutations: {
    SET_STATUS_MESSAGE(state, message) {
      state.sync_state = `${message}`
    },
    SET_SNACKBAR_MESSAGE(state, payload) {
      state.snackbarMessage = payload.snackbarMessage
      state.snackbarColor = payload.snackbarColor

      state.snackbar = true
    },
    SET_SNACKBAR(state, snackbar) {
      state.snackbar = snackbar
    },
    // ADD_MONTH(state) {
    //   state.selected_month = moment(state.selected_month)
    //     .add(1, 'M')
    //     .format('YYYY-MM')
    // },
    // PREVIOUS_MONTH(state) {
    //   state.selected_month = moment(state.selected_month)
    //     .subtract(1, 'M')
    //     .format('YYYY-MM')
    // },
    // GO_TO_CURRENT_MONTH(state) {
    //   state.selected_month = moment(new Date()).format('YYYY-MM')
    // },
    UPDATE_SELECTED_BUDGET(state, selectedBudgetID) {
      state.selectedBudgetID = selectedBudgetID
      localStorage.budgetID = selectedBudgetID
    },
    UPDATE_SELECTED_MONTH(state, year_month) {
      state.selectedMonth = year_month
    }
  },
  actions: {
    setSnackBarBoolean(context, snackbar) {
      context.commit('SET_SNACKBAR', snackbar)
    },
    setSelectedBudgetID(context, payload) {
      context.commit('UPDATE_SELECTED_BUDGET', payload)
      context.dispatch('loadLocalBudgetRoot')
    }
  }
})
