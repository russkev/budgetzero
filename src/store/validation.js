import validateSchema from 'json-schema'
import { ID_LENGTH, ID_NAME } from '../constants'

const account_id_length = 'b_'.length + ID_LENGTH.budget + ID_NAME.account.length + ID_LENGTH.account
const schema_account = {
  type: 'object',
  properties: {
    _id: {
      type: 'string',
      minimum: account_id_length,
      maximum: account_id_length
    },
    onBudget: {
      type: 'boolean'
    },
    name: {
      type: 'string'
    },
    sort: {
      type: 'integer'
    },
    closed: {
      type: 'boolean'
    },
    type: {
      type: 'string'
    },
    csvInfo: {
      type: 'object'
    }
  },
  required: ['_id', 'onBudget', 'name', 'sort', 'closed', 'type']
}

const transaction_id_length = 'b_'.length + ID_LENGTH.budget + ID_NAME.transaction.length + ID_LENGTH.transaction
const schema_transaction = {
  type: 'object',
  properties: {
    value: {
      type: 'integer'
    },
    date: {
      type: 'string'
    },
    category: {
      type: ['string', 'null']
    },
    account: {
      type: 'string'
    },
    memo: {
      type: ['string', 'null']
    },
    cleared: {
      type: 'boolean'
    },
    reconciled: {
      type: 'boolean'
    },
    flag: {
      type: 'string'
    },
    payee: {
      type: ['string', 'null']
    },
    transfer: {
      type: ['string', 'null']
    },
    splits: {
      type: ['null', 'array', 'object']
    },
    balance: {
      type: 'integer'
    },
    note: {
      type: ['string', 'null']
    },
    _id: {
      type: 'string',
      minimum: transaction_id_length,
      maximum: transaction_id_length
    }
  },
  required: [
    'value',
    'date',
    'category',
    'account',
    'memo',
    'cleared',
    'reconciled',
    'flag',
    'payee',
    'transfer',
    'splits',
    '_id'
  ]
}

const category_id_length = 'b_'.length + ID_LENGTH.budget + ID_NAME.category.length + ID_LENGTH.category
const schema_category = {
  type: 'object',
  properties: {
    _id: {
      type: 'string',
      minLength: category_id_length,
      maxLength: category_id_length
    },
    sort: {
      type: ['null', 'integer', 'number']
    },
    masterCategory: {
      type: 'string',
      minLength: ID_LENGTH.category,
      maxLength: ID_LENGTH.category
    },
    name: {
      type: 'string'
    },
    hidden: {
      type: ['boolean', 'null']
    }
  },
  required: ['_id', 'sort', 'masterCategory', 'name']
}

const monthCategory_id_length =
  'b_'.length + ID_LENGTH.budget + `${ID_NAME.monthCategory}2000-01_`.length + ID_LENGTH.category
const schema_monthCategory = {
  type: 'object',
  properties: {
    _id: {
      type: 'string',
      minLength: monthCategory_id_length,
      maxLength: monthCategory_id_length
    },
    budget: {
      type: ['null', 'integer']
    },
    spent: {
      type: ['null', 'integer']
    },

    note: {
      type: ['string', 'null']
    }
  },
  required: ['_id', 'budget', 'overspending']
}

const masterCategory_id_length = 'b_'.length + ID_LENGTH.budget + ID_NAME.masterCategory.length + ID_LENGTH.category
const schema_masterCategory = {
  type: 'object',
  properties: {
    _id: {
      type: 'string',
      minLength: masterCategory_id_length,
      maxLength: masterCategory_id_length
    },
    name: {
      type: 'string'
    },
    sort: {
      type: ['integer', 'number']
    },
    collapsed: {
      type: ['boolean', 'null']
    },
    color: {
      type: ['object', 'null']
    }
  },
  required: ['_id', 'name', 'sort', 'collapsed']
}

const payee_id_length = 'b_'.length + ID_LENGTH.budget + ID_NAME.payee.length + ID_LENGTH.payee
const schema_payee = {
  type: 'object',
  properties: {
    _id: {
      type: 'string',
      minLength: payee_id_length,
      maxLength: payee_id_length
    },
    name: {
      type: 'string'
    }
  },
  required: ['_id', 'name']
}

const budget_id_length = ID_NAME.budget.length + ID_LENGTH.budget
const schema_budget = {
  type: 'object',
  properties: {
    _id: {
      type: 'string',
      minLength: budget_id_length,
      maxLength: budget_id_length
    },
    name: {
      type: 'string'
    },
    created: {
      type: 'string'
    },
    accessed: {
      type: 'number'
    },
    currency: {
      type: 'string'
    },
    checkNumber: {
      type: 'boolean'
    }
  },
  required: ['_id', 'name', 'created', 'accessed', 'currency', 'checkNumber']
}

export {
  schema_budget,
  schema_account,
  schema_transaction,
  schema_category,
  schema_monthCategory,
  schema_masterCategory,
  schema_payee,
  validateSchema
}
