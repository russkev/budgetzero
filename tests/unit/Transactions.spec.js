import { shallowMount, mount, createLocalVue } from "@vue/test-utils";
import Vuex from "vuex";
import Transactions from "@/components/TransactionView/Transactions.vue";
import Vuetify from "vuetify";
import mock_budget from "@/../tests/__mockdata__/mock_budget.json";
import store from "@/store";

import Vue from "vue";
Vue.use(Vuetify);
Vue.config.productionTip = false;

const localVue = createLocalVue();
localVue.use(Vuex);

const $route = {
  path: "/fake/path",
  params: {
    account_id: "38e690f8-198f-4735-96fb-3a2ab15081c2"
  }
};

const data = mock_budget.rows.map(row => row.doc);
let numberOfTransactions = null

describe("transaction table", () => {
  let vuetify;
  let wrapper;

  beforeAll(() => {
    // state.monthCategoryBudgets = data.filter(row => row._id.includes("_monthCategory_"));
    // state.payees = data.filter(row => row._id.includes("_payee_"));

    store.state.pouchdb.accounts = data.filter(row => row._id.includes("_account_"));
    store.state.pouchdb.transactions = data.filter(row => row._id.includes("_transaction_"));
    store.state.pouchdb.masterCategories = data.filter(row =>
      row._id.includes("_master-category_")
    );
    store.state.pouchdb.categories = data
      .filter(row => row._id.includes("_category_"))
      .filter(row => !row._id.includes("monthCategory"));
    // store.state.pouchdb.month_selected = "2020-12",
    // store.state.pouchdb.selectedBudgetID = "79de488f-448e-4b4d-97ad-61e5e4f5df31",
    
    numberOfTransactions = data.filter(row => row._id.includes("_transaction_")).length
  });

  
  beforeEach(() => {
    vuetify = new Vuetify();
    localVue.use(vuetify)

    store.state.month_selected = "2020-12";
    wrapper = shallowMount(Transactions, {
      store,
      localVue,
      vuetify,
      mocks: {
        $route
      },
      stubs: {
        ImportModalComponent: true
      },
    });
  });



  it("renders correct snapshot", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("transaction list for table is correct length", () => {
    expect(wrapper.vm.transactionListForTable.length).toBe(numberOfTransactions);
  });
  
  it("add transaction btn exists", () => {
    expect(wrapper.find("#addTransactionBtn").exists()).toBeTruthy();
  });

  // it("add transaction btn creates a new row when clicked", () => {
  //   expect(wrapper.vm.transactionListForTable.length).toBe(numberOfTransactions);
  //   wrapper.find("#addTransactionBtn").trigger('click')
  //   expect(wrapper.vm.transactionListForTable.length).toBe(numberOfTransactions+1);
  // });
  
});

describe("reconcile function", () => {
  let vuetify;
  let wrapper;

  beforeAll(() => {
    // state.monthCategoryBudgets = data.filter(row => row._id.includes("_monthCategory_"));
    // state.payees = data.filter(row => row._id.includes("_payee_"));

    store.state.pouchdb.accounts = data.filter(row => row._id.includes("_account_"));
    store.state.pouchdb.transactions = data.filter(row => row._id.includes("_transaction_"));
    store.state.pouchdb.masterCategories = data.filter(row =>
      row._id.includes("_master-category_")
    );
    store.state.pouchdb.categories = data
      .filter(row => row._id.includes("_category_"))
      .filter(row => !row._id.includes("monthCategory"));
    // store.state.pouchdb.month_selected = "2020-12",
    // store.state.pouchdb.selectedBudgetID = "79de488f-448e-4b4d-97ad-61e5e4f5df31",

    store.state.route = $route
    
    numberOfTransactions = data.filter(row => row._id.includes("_transaction_")).length
  });

  
  beforeEach(() => {
    vuetify = new Vuetify();
    localVue.use(vuetify)

    store.state.month_selected = "2020-12";
    wrapper = mount(Transactions, {
      store,
      localVue,
      vuetify,
      mocks: {
        $route
      },
      stubs: {
        ImportModalComponent: true
      },
    });
  });

  it("reconcile button exists", () => {
    expect(wrapper.find('#btn-reconcile').exists()).toBeTruthy();
  });

  it("reconcile button exists", () => {
    wrapper.find('#btn-reconcile').trigger('click')
    
  });


});