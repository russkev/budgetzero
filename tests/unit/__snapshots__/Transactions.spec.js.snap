// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transaction table renders correct snapshot 1`] = `
<v-container-stub tag="div" fluid="true" fill-height="" class="ma-0 pa-0">
  <basedialogmodalcomponent-stub></basedialogmodalcomponent-stub>
  <transactionheader-stub selected_account="[object Object]"></transactionheader-stub>
  <!---->
  <v-divider-stub></v-divider-stub>
  <v-row-stub tag="div" class="transaction-toolbar ma-0">
    <v-col-stub tag="div" class="pt-2 pb-0">
      <v-btn-stub color="accent darken-1" dark="true" outlined="true" tag="button" activeclass="" small="true" type="button" id="addTransactionBtn" class="mb-2">
        <v-icon-stub color="accent" left="" medium=""> mdi-plus </v-icon-stub>
        Transaction
      </v-btn-stub>
    </v-col-stub>
    <v-col-stub tag="div" class="py-1 body-1 font-weight-bold">
      <v-menu-stub opendelay="0" closedelay="0" bottom="true" openonclick="true" contentclass="" maxwidth="auto" nudgebottom="0" nudgeleft="0" nudgeright="0" nudgetop="0" nudgewidth="0" closeonclick="true" closeoncontentclick="true" maxheight="auto" offsetx="true" origin="top left" transition="v-menu-transition">
        <v-list-stub tag="div">
          <v-list-item-stub activeclass="" tag="div">
            <v-icon-stub dark="" color="primary" left=""> mdi-check-circle </v-icon-stub>
            <v-list-item-title-stub tag="div">Approve</v-list-item-title-stub>
          </v-list-item-stub>
          <v-list-item-stub activeclass="" tag="div">
            <v-icon-stub dark="" color="primary" left=""> mdi-alpha-c-circle </v-icon-stub>
            <v-list-item-title-stub tag="div">Clear</v-list-item-title-stub>
          </v-list-item-stub>
          <v-list-item-stub activeclass="" tag="div">
            <v-icon-stub dark="" color="red" left=""> mdi-delete </v-icon-stub>
            <v-list-item-title-stub tag="div">Delete</v-list-item-title-stub>
          </v-list-item-stub>
          <v-menu-stub opendelay="0" closedelay="0" bottom="true" openonclick="true" openonhover="true" contentclass="" maxwidth="auto" nudgebottom="0" nudgeleft="0" nudgeright="0" nudgetop="0" nudgewidth="0" closeonclick="true" closeoncontentclick="true" maxheight="auto" offsetx="true" origin="top left" transition="v-menu-transition">
            <v-list-stub tag="div">
              <v-list-item-stub activeclass="" tag="div">
                <v-list-item-title-stub tag="div"> Birthdays </v-list-item-title-stub>
              </v-list-item-stub>
              <v-list-item-stub activeclass="" tag="div">
                <v-list-item-title-stub tag="div"> Cable TV </v-list-item-title-stub>
              </v-list-item-stub>
              <v-list-item-stub activeclass="" tag="div">
                <v-list-item-title-stub tag="div"> Car Insurance </v-list-item-title-stub>
              </v-list-item-stub>
              <v-list-item-stub activeclass="" tag="div">
                <v-list-item-title-stub tag="div"> Car Maintenance </v-list-item-title-stub>
              </v-list-item-stub>
              <v-list-item-stub activeclass="" tag="div">
                <v-list-item-title-stub tag="div"> Car Payment </v-list-item-title-stub>
              </v-list-item-stub>
              <v-list-item-stub activeclass="" tag="div">
                <v-list-item-title-stub tag="div"> Car Replacement </v-list-item-title-stub>
              </v-list-item-stub>
              <v-list-item-stub activeclass="" tag="div">
                <v-list-item-title-stub tag="div"> Charitable </v-list-item-title-stub>
              </v-list-item-stub>
              <v-list-item-stub activeclass="" tag="div">
                <v-list-item-title-stub tag="div"> Christmas </v-list-item-title-stub>
              </v-list-item-stub>
              <v-list-item-stub activeclass="" tag="div">
                <v-list-item-title-stub tag="div"> Clothing </v-list-item-title-stub>
              </v-list-item-stub>
              <v-list-item-stub activeclass="" tag="div">
                <v-list-item-title-stub tag="div"> Electricity </v-list-item-title-stub>
              </v-list-item-stub>
              <v-list-item-stub activeclass="" tag="div">
                <v-list-item-title-stub tag="div"> Emergency Fund </v-list-item-title-stub>
              </v-list-item-stub>
              <v-list-item-stub activeclass="" tag="div">
                <v-list-item-title-stub tag="div"> Fuel </v-list-item-title-stub>
              </v-list-item-stub>
              <v-list-item-stub activeclass="" tag="div">
                <v-list-item-title-stub tag="div"> Groceries </v-list-item-title-stub>
              </v-list-item-stub>
              <v-list-item-stub activeclass="" tag="div">
                <v-list-item-title-stub tag="div"> Household Goods </v-list-item-title-stub>
              </v-list-item-stub>
              <v-list-item-stub activeclass="" tag="div">
                <v-list-item-title-stub tag="div"> Income Next Month </v-list-item-title-stub>
              </v-list-item-stub>
              <v-list-item-stub activeclass="" tag="div">
                <v-list-item-title-stub tag="div"> Income This Month </v-list-item-title-stub>
              </v-list-item-stub>
              <v-list-item-stub activeclass="" tag="div">
                <v-list-item-title-stub tag="div"> Internet </v-list-item-title-stub>
              </v-list-item-stub>
              <v-list-item-stub activeclass="" tag="div">
                <v-list-item-title-stub tag="div"> Medical/Dental </v-list-item-title-stub>
              </v-list-item-stub>
              <v-list-item-stub activeclass="" tag="div">
                <v-list-item-title-stub tag="div"> Phone </v-list-item-title-stub>
              </v-list-item-stub>
              <v-list-item-stub activeclass="" tag="div">
                <v-list-item-title-stub tag="div"> Rainy Day Funds </v-list-item-title-stub>
              </v-list-item-stub>
              <v-list-item-stub activeclass="" tag="div">
                <v-list-item-title-stub tag="div"> Rent/Mortgage </v-list-item-title-stub>
              </v-list-item-stub>
              <v-list-item-stub activeclass="" tag="div">
                <v-list-item-title-stub tag="div"> Renters Insurance </v-list-item-title-stub>
              </v-list-item-stub>
              <v-list-item-stub activeclass="" tag="div">
                <v-list-item-title-stub tag="div"> Restaurants </v-list-item-title-stub>
              </v-list-item-stub>
              <v-list-item-stub activeclass="" tag="div">
                <v-list-item-title-stub tag="div"> Retirement </v-list-item-title-stub>
              </v-list-item-stub>
              <v-list-item-stub activeclass="" tag="div">
                <v-list-item-title-stub tag="div"> Spending Money </v-list-item-title-stub>
              </v-list-item-stub>
              <v-list-item-stub activeclass="" tag="div">
                <v-list-item-title-stub tag="div"> Student Loan Payment </v-list-item-title-stub>
              </v-list-item-stub>
              <v-list-item-stub activeclass="" tag="div">
                <v-list-item-title-stub tag="div"> Tithing </v-list-item-title-stub>
              </v-list-item-stub>
              <v-list-item-stub activeclass="" tag="div">
                <v-list-item-title-stub tag="div"> Uncategorized </v-list-item-title-stub>
              </v-list-item-stub>
              <v-list-item-stub activeclass="" tag="div">
                <v-list-item-title-stub tag="div"> Vacation </v-list-item-title-stub>
              </v-list-item-stub>
              <v-list-item-stub activeclass="" tag="div">
                <v-list-item-title-stub tag="div"> Water </v-list-item-title-stub>
              </v-list-item-stub>
            </v-list-stub>
          </v-menu-stub>
        </v-list-stub>
      </v-menu-stub>
      <v-tooltip-stub opendelay="0" closedelay="0" bottom="true" openonclick="true" openonhover="true" openonfocus="true" contentclass="" maxwidth="auto" nudgebottom="0" nudgeleft="0" nudgeright="0" nudgetop="0" nudgewidth="0" tag="span"> <span>Delete</span></v-tooltip-stub>
      <v-btn-stub color="grey darken-1" tag="button" activeclass="" icon="true" type="button" class="white--text">
        <v-icon-stub> mdi-refresh </v-icon-stub>
      </v-btn-stub>
    </v-col-stub>
    <v-col-stub tag="div" class="pt-2 pb-0">
      <v-btn-stub color="grey lighten-2" elevation="0" tag="button" activeclass="" small="true" type="button" class="mb-2 ml-2">
        <v-icon-stub left=""> mdi-cloud-upload </v-icon-stub> <span>Import</span>
      </v-btn-stub>
    </v-col-stub>
    <v-col-stub tag="div" class="pt-0 pb-0">
      <v-text-field-stub errorcount="1" errormessages="" messages="" rules="" successmessages="" value="" appendicon="search" backgroundcolor="" hidedetails="true" label="Search" loaderheight="2" clearable="true" clearicon="$clear" singleline="true" type="text" class="pt-0"></v-text-field-stub>
    </v-col-stub>
  </v-row-stub>
  <v-divider-stub></v-divider-stub>
  <v-row-stub tag="div" class="ma-0">
    <v-col-stub tag="div" class="pa-0 ma-0">
      <importmodalcomponent-stub account="38e690f8-198f-4735-96fb-3a2ab15081c2"></importmodalcomponent-stub>
      <v-form-stub>
        <v-data-table-stub mobilebreakpoint="600" items="[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]" options="[object Object]" sortby="date" sortdesc="true" customsort="[Function]" page="1" itemsperpage="10" groupby="" groupdesc="" customgroup="[Function]" locale="en-US" customfilter="[Function]" serveritemslength="-1" itemkey="_id" value="" expanded="" singleexpand="true" noresultstext="$vuetify.dataIterator.noResultsText" nodatatext="$vuetify.noDataText" loadingtext="$vuetify.dataIterator.loadingText" footerprops="[object Object]" selectablekey="isSelectable" loaderheight="4" headers="[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]" showselect="true" height="573" fixedheader="true" expandicon="$expand" itemclass="" class="elevation-1 transaction-table"></v-data-table-stub>
      </v-form-stub>
    </v-col-stub>
  </v-row-stub>
</v-container-stub>
`;
